testing/trino-server-dev/src/main/java/io/trino/server/HybridDevelopmentPluginsProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server;
+
+import com.google.inject.Inject;
+import io.trino.server.PluginManager.PluginsProvider;
+
+import static java.util.Objects.requireNonNull;
+
+public class HybridDevelopmentPluginsProvider
+        implements PluginsProvider
+{
+    private final DevelopmentPluginsProvider developmentPluginsProvider;
+    private final ServerPluginsProvider serverPluginsProvider;
+
+    @Inject
+    public HybridDevelopmentPluginsProvider(DevelopmentPluginsProvider developmentPluginsProvider, ServerPluginsProvider serverPluginsProvider)
+    {
+        this.developmentPluginsProvider = requireNonNull(developmentPluginsProvider, "developmentPluginsProvider is null");
+        this.serverPluginsProvider = requireNonNull(serverPluginsProvider, "serverPluginsProvider is null");
+    }
+
+    @Override
+    public void loadPlugins(Loader loader, ClassLoaderFactory createClassLoader)
+    {
+        developmentPluginsProvider.loadPlugins(loader, createClassLoader);
```

Comment:
This looks ok to me, but I wonder if we can improve the developer experience here. Coping plugins is one extra step that could be avoided - if the `DevelopmentPluginsProvider` was able to load a plugin from a directory, then we could add directories to the `plugin.bundles` list.

Reply:
Hm I see what you're saying. Currently, `plugins.bundle` supports `pom.xml`/`*.pom` files as well as Maven coordinates, and we could _also_ add support for directories. I can take a look at this.